### PR TITLE
[Feature] Support Primary Key table Backup & Restore

### LINKED ISSUE #11879: [Feature] Support Primary Key table Backup & Restore
## Feature request

Problem:
At present, StarRocks does not support the backup and restore of Primary Key tables. This feature will support it.

Solution:
We basically reused the code of the primary key clone and transformed it to adapt to the backup restore process.

### PR DESCRIPTION
**Pull request by srlch**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #11879

## Problem Summary(Required) ：
[Feature] Support Primary Key table Backup & Restore(#11879)

Problem:
Currently, StarRocks does not support the backup and restore of Primary Key tables. This feature will support it.

Solution:
We reused the code of the primary-key clone and transformed it to adapt to the backup restore process.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|0|1|00.00%|[82]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 3 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|0|1|00.00%|[82]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|0|2|00.00%|[625, 626]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|0|1|00.00%|[82]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|0|1|00.00%|[82]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

**Review by decster — COMMENTED** [review]
if is_restore_task==true, just create dir, do not go through the whole make_snapshot process, seems easier.

File: be/src/storage/tablet.cpp [hunk 1]
```diff
@@ -526,8 +526,13 @@ void Tablet::list_versions(vector<Version>* versions) const {
     }
 }
 
-Status Tablet::capture_consistent_rowsets(const Version& spec_version, std::vector<RowsetSharedPtr>* rowsets) const {
+Status Tablet::capture_consistent_rowsets(const Version& spec_version, std::vector<RowsetSharedPtr>* rowsets,
+                                          bool ingnore) const {
```
[the remark this answers is not in the document]
Comment (srlch) [reply be/src/storage/tablet.cpp]: have fixed it.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

**Review by decster — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**